autotest/ogr/ogr_mssqlspatial.py

```diff
@@ -711,6 +736,35 @@ def test_binary_field_bcp():
 #
 
 
+def test_geometry_column_identification():
+    """Test for issue GH #1190
+    SQL Server data source: geometry column not identified by ogr2ogr"""
+
+    if gdaltest.mssqlspatial_ds is None:
+        pytest.skip("MSSQLSpatial driver not available")
+
+    gdaltest.mssqlspatial_ds.ExecuteSQL(
+        "SELECT *, ogr_geometry.STCentroid() AS shape INTO dbo.poly_shape FROM dbo.tpoly"
+    )
```

Comment:
is it useful ? The result is not tested.
```suggestion
    gdaltest.mssqlspatial_ds.ExecuteSQL(
        "SELECT *, ogr_geometry.STCentroid() AS shape INTO dbo.poly_shape FROM dbo.tpoly"
    )
```

Furthermore, the result of ExecuteSQL() (when it returns something) must be explicitly freed with ds.ReleaseResultSet(lyr_returned_by_executeSql)
A way of doing that introduced recently is to use
```
with ds.ExecuteSQL(...) as lyr:
     .. do something ..
```

Reply:
that's a preparation step that creates a new table with SELECT INTO since you mentioned to not alter tpoly which is used by another test case.

Reply:
as sorry, I missed the INTO ... . I thought it was a simple SELECT. Does it return None or non-None ? If non-None, that should be freed

Reply:
it returns None